dev-utils/visualise-ui-hierarchy/tree2dot

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+import sys
+import fileinput
+from enum import Enum
+
+
+class Node:
+    class Type(Enum):
+        SOURCE = "red"
+        POLY_LOOK = "green"
+        THIRD_PARTY = "blue"
+
+    def __init__(self, key, value, type):
+        self.key = key
+        self.value = value
+        self.type = type
+
+
+class Graph:
+    def __init__(self):
+        self.__nodes = {}
+        self.edges = set()
+
+    def register_node(self, value, type):
+        if value not in self.__nodes:
+            key = str(len(self.__nodes) + 1)
+            self.__nodes[value] = Node(key, value, type)
+        return self.__nodes[value]
+
+    def connect(self, from_value, to_value):
+        from_key = self.__nodes[from_value].key
+        to_key = self.__nodes[to_value].key
+        self.edges.add((from_key, to_key))
```

Comment:
You need a numeric value to index every node, right?

Reply:
Can be an arbitrary string really, but I didn't want to use those massive paths as keys - it's a bit ugly that way in the dot file.